csharp/src/Ice/ConnectionI.cs

```diff
@@ -2024,48 +2037,22 @@ private int sendMessage(OutgoingMessage message)
         {
             cancelInactivityTimer();
         }
-        // If we're sending a heartbeat, there is a chance the connection is inactive and that we need to schedule the
-        // inactivity timer. It's ok to do this before actually sending the heartbeat since the heartbeat does not count
-        // as an "activity".
-        else if (
-            _inactivityTimer is null &&           // timer not already scheduled
-            _inactivityTimeout > TimeSpan.Zero && // inactivity timeout is enabled
-            _state == StateActive &&              // only schedule the timer if the connection is active
-            _dispatchCount == 0 &&                // no pending dispatch
-            _asyncRequests.Count == 0 &&          // no pending invocation
-            _readHeader)                          // we're not waiting for the remainder of an incoming message
-        {
-            bool isInactive = true;
 
-            // We may become inactive while the peer is back-pressuring us. In this case, we only schedule the
-            // inactivity timer if all outgoing messages in _sendStreams are heartbeats.
-            foreach (OutgoingMessage queuedMessage in _sendStreams)
+        if (_sendStreams.Count > 0)
+        {
+            if (isHeartbeat)
```

Comment:
Why not just move the check to `sendHeartbeat`? Don't call `sendMessage` if `_sendStreams.Count > 0`.

Reply:
Moved.